applications/OptimizationApplication/python_scripts/controls/control.py

```diff
@@ -0,0 +1,114 @@
+        in their design space.
+
+        If the case is that this control has one physical variable, then it should be same as the control variable.
+
+        Returns:
+            list[SupportedSensitivityFieldVariableTypes]: List of physical control variables.
+        """
+        pass
+
+    @abstractmethod
+    def GetEmptyControlField(self) -> ContainerExpressionTypes:
+        """Returns a new empty control field data holder.
+
+        This returns a new empty control field data holder to give information about on which model part's container
+        this model part is acting on. This has O(1) complexity, hence has the least cost because it does not read
+        any data from respective model part's container.
+
+        Returns:
+            ContainerExpressionTypes: Returns a new empty ContainerExpression corresponding to control's model part's respective container.
+        """
+        pass
+
+    @abstractmethod
+    def MapGradient(self, physical_gradient_variable_container_expression_map: 'dict[SupportedSensitivityFieldVariableTypes, ContainerExpressionTypes]') -> ContainerExpressionTypes:
+        """Maps physical space gradients to the control space.
+
+        This method is used to map the given physical space gradients to the control space. The input should be as in the following example:
+            physical_gradient_variable_container_expression_map = {
+                Kratos.YOUNG_MODULUS: Kratos.ContainerExpressions.NodalNonHistoricalContainer,
+                Kratos.DENSITY      : Kratos.ContainerExpressions.ElementNonHistoricalContainer,
+                Kratos.SHAPE        : Kratos.ContainerExpressions.ElementNonHistoricalContainer
+            }
+
+        All the gradients w.r.t. @see GetPhysicalKratosVariables() variables will be given in @ref physical_gradient_variable_container_expression_map.
+        If the response does not depend on some of them or all, then container expressions with ContainerExpression::SetDataToZero() performed will be passed.
+        [ContainerExpression::SetDataToZero() will make the container expression to zero, but will only allocate one double value for the
+        whole container, hence it is cheap in memory and execution.]
+
+        Args:
+            physical_gradient_variable_container_expression_map (dict[SupportedSensitivityFieldVariableTypes, ContainerExpressionTypes]): Map of physical space variable and ContainerExpression with sensitivities.
+
+        Returns:
+            ContainerExpressionTypes: Gradients mapped in to control space.
+        """
+        pass
+
+    @abstractmethod
+    def Update(self, control_field: ContainerExpressionTypes) -> bool:
+        """Modifies the current control with the given control field.
+
+        This method modifies the control with the given control field from @ref control_field. This
+        control_field should be in the control space, and it should not represent a change of design.
```

Comment:
This is extra info with the assumption that there is always two spaces, namely physical and control. Please remove

Reply:
done :)